From 8df4bc3ee67f6cc03b1a149264f1f806d5fe2ef7 Mon Sep 17 00:00:00 2001
From: Brahmajit Das <listout@listout.xyz>
Date: Mon, 10 Aug 2026 22:04:59 +0530
Subject: [PATCH] From dd0a7e07f253248de4f0a5da5547e0534438f569 Mon Sep 17
 00:00:00 2001 Message-ID:
 <dd0a7e07f253248de4f0a5da5547e0534438f569.1743862811.git.mprivozn@redhat.com>
 From: Michal Privoznik <mprivozn@redhat.com> Date: Tue, 15 Mar 2022 05:23:29
 +0100 Subject: [PATCH] libvirt-11.3.0-fix-paths-for-apparmor.patch

From: Michal Privoznik <mprivozn@redhat.com>

Rebased for 12.6.0

Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
Signed-off-by: Brahmajit Das <listout@listout.xyz>
---
 src/security/apparmor/libvirt-qemu                              | 1 +
 src/security/apparmor/meson.build                               | 2 +-
 src/security/apparmor/usr.lib.libvirt.virt-aa-helper.local      | 1 -
 ....virt-aa-helper.in => usr.libexec.libvirt.virt-aa-helper.in} | 2 +-
 src/security/apparmor/usr.libexec.libvirt.virt-aa-helper.local  | 1 +
 5 files changed, 4 insertions(+), 3 deletions(-)
 delete mode 100644 src/security/apparmor/usr.lib.libvirt.virt-aa-helper.local
 rename src/security/apparmor/{usr.lib.libvirt.virt-aa-helper.in => usr.libexec.libvirt.virt-aa-helper.in} (96%)
 create mode 100644 src/security/apparmor/usr.libexec.libvirt.virt-aa-helper.local

diff --git a/src/security/apparmor/libvirt-qemu b/src/security/apparmor/libvirt-qemu
index e4aceacd7062..6ac5149bafe8 100644
--- a/src/security/apparmor/libvirt-qemu
+++ b/src/security/apparmor/libvirt-qemu
@@ -97,6 +97,7 @@
   /usr/share/sgabios/** r,
   /usr/share/slof/** r,
   /usr/share/vgabios/** r,
+  /usr/share/seavgabios/** r,
 
   # pki for libvirt-vnc and libvirt-spice (LP: #901272, #1690140)
   /etc/pki/CA/ r,
diff --git a/src/security/apparmor/meson.build b/src/security/apparmor/meson.build
index 7585fb710048..1bae08c2643c 100644
--- a/src/security/apparmor/meson.build
+++ b/src/security/apparmor/meson.build
@@ -1,5 +1,5 @@
 apparmor_gen_profiles = [
-  'usr.lib.libvirt.virt-aa-helper',
+  'usr.libexec.libvirt.virt-aa-helper',
   'usr.sbin.virtqemud',
   'usr.sbin.virtxend',
 ]
diff --git a/src/security/apparmor/usr.lib.libvirt.virt-aa-helper.local b/src/security/apparmor/usr.lib.libvirt.virt-aa-helper.local
deleted file mode 100644
index c0990e51d0b5..000000000000
--- a/src/security/apparmor/usr.lib.libvirt.virt-aa-helper.local
+++ /dev/null
@@ -1 +0,0 @@
-# Site-specific additions and overrides for 'usr.lib.libvirt.virt-aa-helper'
diff --git a/src/security/apparmor/usr.lib.libvirt.virt-aa-helper.in b/src/security/apparmor/usr.libexec.libvirt.virt-aa-helper.in
similarity index 96%
rename from src/security/apparmor/usr.lib.libvirt.virt-aa-helper.in
rename to src/security/apparmor/usr.libexec.libvirt.virt-aa-helper.in
index 1f1b80b9f98e..a4dcb6836a3a 100644
--- a/src/security/apparmor/usr.lib.libvirt.virt-aa-helper.in
+++ b/src/security/apparmor/usr.libexec.libvirt.virt-aa-helper.in
@@ -75,5 +75,5 @@ profile virt-aa-helper @libexecdir@/virt-aa-helper {
   /**.[iI][sS][oO] r,
   /**/disk{,.*} r,
 
-  include if exists <local/usr.lib.libvirt.virt-aa-helper>
+  include if exists <local/usr.libexec.libvirt.virt-aa-helper>
 }
diff --git a/src/security/apparmor/usr.libexec.libvirt.virt-aa-helper.local b/src/security/apparmor/usr.libexec.libvirt.virt-aa-helper.local
new file mode 100644
index 000000000000..974653d7973a
--- /dev/null
+++ b/src/security/apparmor/usr.libexec.libvirt.virt-aa-helper.local
@@ -0,0 +1 @@
+# Site-specific additions and overrides for 'usr.libexec.libvirt.virt-aa-helper'
-- 
2.55.0

